src/views/ProjectView.vue

```diff
@@ -26,85 +27,99 @@ const noSelectedImages = computed(() => {
   return store.getters.selectedImages.length === 0
 })
 
+// manages successful api response by mapping data to unique sessions
+const handleSuccess = (data) => {
+    const results = data.results
+    uniqueDataSessions.value = results
+    .map(session => ({ id: session.id, name: session.name }))
+    isPopupVisible.value = true
+}
+
+// manages api call failures by logging errors
+const handleError = (error) => {
+    console.error('API call failed with error:', error)
+    errorMessage.value = error.message || 'An error occurred'
+}
+
+// fetches session data from API and handles response or error using the callbacks
 const getDataSessions = async () => {
-    const url = 'http://127.0.0.1:8000/api/datasessions/'
     try {
-        const data = await fetchApiCall(url, 'GET')
-        const results = data.results
-        const uniqueNames = new Set()
-
-        uniqueDataSessions.value = results
-        .filter(session => {
-            const isUnique = !uniqueNames.has(session.name) && !uniqueNames.has(session.id)
-            uniqueNames.add(session.name)
-            uniqueNames.add(session.id)
-            return isUnique
-        })
-        .map(session => ({ id: session.id, name: session.name }))
-        isPopupVisible.value = true
+        await fetchApiCall({ url: apiBaseUrl, method: 'GET', successCallback: handleSuccess, failCallback: handleError })
     } catch (error) {
-        console.error('Error getting data sessions: ', error)
+        handleError(error)
     }
 }
 
+// updates an existing session with selected images
 const addImagesToExistingSession = async (session) => {
-    const selectedImages = store.state.selectedImages
-    const inputData = selectedImages.map(image => ({
-      'source': image.image,
-      'basename': image.basefile_name
-    }))
+    const sessionIdUrl = apiBaseUrl + session.id + '/'
+    try {
+        // fetches existing session data
+        const currentSessionResponse = await fetchApiCall({ url: sessionIdUrl, method: 'GET' })
+        const currentSessionData = currentSessionResponse.input_data
 
-    const requestBody = {
-      'name': session.name,
-      'input_data': inputData
-    }
+        // merging existing and new image data
```

Comment:
Maybe add a comment with a TODO that this is temporary until we change it in the backend?